src/ui/Base/RegisteredNamedMethods.ts

```diff
@@ -447,6 +448,68 @@ Initialization.registerNamedMethod('configureResourceRoot', (path: string) => {
   configureResourceRoot(path);
 });
 
+function findAllComponentsOfType<T extends BaseComponent>(type: { new (...args: any[]): T }, searchRoot = document.body) {
+  const className = Component.computeCssClassNameForType((<typeof BaseComponent>(<{}>type)).ID);
+  const root = $$(searchRoot);
+  const found: T[] = [];
+  if (root.hasClass(className)) {
+    found.push(<T>Component.resolveBinding(root.el, type));
+  }
+  found.push(...root.findClass(className).map(element => <T>Component.resolveBinding(element as HTMLElement, type)));
+  return found;
+}
+
+/**
+ * Re-enables all [Analytics]{@link Analytics} components if they were previously disabled.
+ * @param searchRoot What ancestor or CoveoAnalytics element to search from in the DOM.
+ */
+export function enableAnalytics(searchRoot = document.body) {
+  const analyticsInstances = findAllComponentsOfType(Analytics, searchRoot);
+  if (analyticsInstances.length === 0) {
+    console.warn('Could not enable analytics because they could not be found.');
```

Comment:
Use Logger here too.  You'll need to import the class and instantiate it.
Btw, we use logger for all logging because we/the clients can enable/disable all logging or only certain logging levels on command.

Reply:
Added a logger.